src/ruby/puppetserver-lib/puppet/server/compiler.rb

```diff
@@ -14,11 +14,40 @@ def compile(request_data)
         node = create_node(processed_hash)
 
         catalog = Puppet::Parser::Compiler.compile(node, processed_hash['job_id'])
+
+        maybe_save(processed_hash, node.facts, catalog)
```

Comment:
If the `maybe_save` function fails, will the catalog not be returned in the response? I'm not sure if failure to save warrants returning an error in the response, but, I'm not too opinionated here; just seems like if we have a catalog that is ready to be `to_data_hash`-ed, it's shame to just throw that catalog away if `maybe_save` fails.

Reply:
I know that the pdb terminus will raise if there's an issue saving, I _think_ that Puppet will raise if a terminus isn't installed, but will double check. My impression was that this was the closest to existing Puppet behavior, while following the same error messages that we give for `find`. Even if this is consistent with existing behavior, I don't know (or have an opinion on) what desired behavior should be.